### PR TITLE
base: fix for docking bug, a.k.a. "Your reputation stinks" (fixed 12.10.20)

### PR DESCRIPTION
**Pull request by dd84ai**
Fix to pob docking problem https://github.com/DiscoveryGC/FLHook/issues/47
It prevents playercntl plugin interference, which incorrect worked caused "your reputation stinks" and etc during pob dock randomly

### CODE REVIEW

**Comment by dd84ai** [comment]
wait a second, it has not solved problems yet

**Comment by dd84ai** [comment]
now it is working regardless of which plugin is loaded first. done,

**Review by dsyalex — CHANGES_REQUESTED** [review]
please squash your commits and add a reasonable description

**Comment by dd84ai** [comment]
Resolved

**Comment by dd84ai** [comment]
Discovered it can be coded possibly easier than I did

**Comment by dd84ai** [comment]
Remade removing not optimized trash ;b

File: Source/FLHookPluginSDK/headers/plugin.h [hunk 1]
```diff
@@ -328,6 +329,12 @@ struct CUSTOM_BASE_IS_DOCKED_STRUCT
 	uint iDockedBaseID;
 };
 
+struct CUSTOM_BASE_IS_IT_POB_STRUCT
+{
+	uint iBase;
+	bool iAnswer;
```
Comment (dsyalex) [inline Source/FLHookPluginSDK/headers/plugin.h:335]:
```suggestion
	bool bAnswer;
```

File: Plugins/Public/base_plugin/Main.cpp [hunk 2]
```diff
@@ -2497,6 +2497,17 @@ void Plugin_Communication_CallBack(PLUGIN_MESSAGE msg, void* data)
...
 		}
 	}
+	if (msg == CUSTOM_IS_IT_POB)
+	{
+		CUSTOM_BASE_IS_IT_POB_STRUCT* info = reinterpret_cast<CUSTOM_BASE_IS_IT_POB_STRUCT*>(data);
+		PlayerBase *base = GetPlayerBase(info->iBase);
+		returncode = SKIPPLUGINS;
+		if (base)
+		{
+			returncode = SKIPPLUGINS;
```
Comment (dsyalex) [inline Plugins/Public/base_plugin/Main.cpp:2507]:
```suggestion
```

**Review by dsyalex — COMMENTED** [review]
What about the -0.55 check on POBs? Players shouldn't be able to dock if their reputation with the base's affiliation is too close to hostile right?

**Comment by dd84ai** [comment]
> What about the -0.55 check on POBs? Players shouldn't be able to dock if their reputation with the base's affiliation is too close to hostile right?

This restriction is supposed to be applied only during defensemode == 2, when base allows docking rights from player reputation
Perhaps sometime later, someone could fix it. It is not a really important feature anyway, too overspecialized.
There would be needed changes in IsDockingAllowed function

File: Source/FLHookPluginSDK/headers/plugin.h (hunk 1 above)
Comment (dd84ai) [reply Source/FLHookPluginSDK/headers/plugin.h]: applied

File: Plugins/Public/base_plugin/Main.cpp (hunk 2 above)
Comment (dd84ai) [reply Plugins/Public/base_plugin/Main.cpp]: applied

**Review by dsyalex — APPROVED** [review]

**Pull request merged**